Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1017 @@
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No  Front Door(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+            $name = $_.ResourceId.Split('/')[8]
+            $resFrontDoor = Get-AzFrontDoor -Name $name -ErrorAction SilentlyContinue
+            $FrontDoorResources = $FrontDoorResources + $resFrontDoor
+            }
+            catch
+            {
+            Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping the Resource: [$($_.ResourceName)]..."
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+    # Fetch all front doors in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "Fetching all Front Doors in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            # Get all Front Doors in a Subscription
+            $FrontDoorResources = Get-AzFrontDoor -ErrorAction Stop
+        }
+        else
+        {
+            if (-not (Test-Path -Path $FilePath))
+            {
+                Write-Host "Input file [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                return
+            }
+
+            Write-Host "Fetching all Front Doors from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $frontdoorDetails = Import-Csv -LiteralPath $FilePath
+            $validFrontDoorDetails = $frontdoorDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+            
+            $validFrontDoorDetails | ForEach-Object {
+                $resourceId = $_.ResourceId
+
+                try
+                {
+                    $frontDoorResource = Get-AzFrontDoor -ErrorAction SilentlyContinue
```

Comment:
This command will give all front doors in subscription, and we are calling in this in loop of all front doors. This will add duplicate front doors.

Reply:
It is already fixed but this wasn't getting updated maybe sync issue.